hoomd/md/EvaluatorPairALJ.h

```diff
@@ -555,7 +574,7 @@ template<unsigned int ndim> class EvaluatorPairALJ
                           shape_j->rounding_radii,
                           shape_i->has_rounding,
                           shape_j->has_rounding);
-                assert(success && !overlap);
+                // assert(success && !overlap);
```

Comment:
I left this assertion here intentionally since it's a handy way to detect when something is wrong. If a simulation crashes and it appears to be a bug in the overlap calculation (like the one that @jekw and @thiv89 fixed somewhat recently), recompiling in debug mode and running would immediately tell you that if this assertion failed. I think it's better to just add a comment as to why this is useful.
```suggestion
                // Unphysical ALJ simulation results may be the result of
                // invalid collision detection from GJK, which will normally
                // occur silently. This assertion helps debug such errors by
                // failing fast in debug mode if GJK failed to converge.
                assert(success && !overlap);
```